source/src/BNFC/Backend/Java/CFtoJLex15.hs

```diff
@@ -45,6 +45,13 @@ import BNFC.Utils (cstring)
 import BNFC.Backend.Common.NamedVariables
 import Text.PrettyPrint
 
+-- This is for now the workaround to differentiate JLex and JFlex.
```

Comment:
I see the point about readability but given the small differences between the two versions, I don't think having two different functions would be any better... You could pass the `JavaLexerParser` value if you prefer (although that means raising an error on `Antlr4`)

Or maybe create a datatype that encapsulate the difference between the two variants? (something like `data JLexVariant = JLexVariant { bufferVar :: String, escapeChars :: [Char], ... }`)